src/main/Config.h

```diff
@@ -37,24 +37,39 @@ enum class ValidationThresholdLevels : int
     ALL_REQUIRED = 2
 };
 
-class Config : public std::enable_shared_from_this<Config>
+enum class ValidatorQuality : int
 {
-    enum class ValidatorQuality : int
-    {
-        VALIDATOR_LOW_QUALITY = 0,
-        VALIDATOR_MED_QUALITY = 1,
-        VALIDATOR_HIGH_QUALITY = 2,
-        VALIDATOR_CRITICAL_QUALITY = 3
-    };
+    VALIDATOR_LOW_QUALITY = 0,
+    VALIDATOR_MED_QUALITY = 1,
+    VALIDATOR_HIGH_QUALITY = 2,
+    VALIDATOR_CRITICAL_QUALITY = 3
+};
 
-    struct ValidatorEntry
-    {
-        std::string mName;
-        std::string mHomeDomain;
-        ValidatorQuality mQuality;
-        PublicKey mKey;
-        bool mHasHistory;
-    };
+struct ValidatorEntry
+{
+    std::string mName;
+    std::string mHomeDomain;
+    ValidatorQuality mQuality;
+    PublicKey mKey;
+    bool mHasHistory;
+};
+
+// This struct holds information necessary to compute the weight of a validator
+// for leader election
+struct ValidatorWeightConfig
+{
+    // Mapping from node ids to info about each validator
+    UnorderedMap<NodeID, ValidatorEntry> mValidatorEntries;
+
+    // Mapping from org names to the number of validators in that org
+    UnorderedMap<std::string, size_t> mHomeDomainSizes;
+
+    // Highest quality level of any validator in the config
+    ValidatorQuality mHighestQuality;
```

Comment:
we need to initialize this enum, otherwise it's not safe to use in the max function below (UB)

Reply:
Good catch. This enum is no longer in the struct with the change to bring the implementation in line with the design doc.